libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -45,6 +49,7 @@ export class BookSearchComponent implements OnInit {
 
   addBookToReadingList(book: Book) {
     this.store.dispatch(addToReadingList({ book }));
+    this.openSnackBar(book);
```

Comment:
So, here, you are opening the snackbar simultaneously with dispatching the `addToReadingList` action. What happens if the `addToReadingList` is not confirmed, and it fails? Then the snackbar would allow the user to "undo" an action that did not succeed in the first place. You should make sure that `addToReadingList` is confirmed before allowing the user to "undo" it.